pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -118,6 +124,8 @@
   private volatile long _lastIndexedTimeMs = Long.MIN_VALUE;
   private volatile long _latestIngestionTimeMs = Long.MIN_VALUE;
 
+  private RealtimeLuceneReadersForRealtimeSegment _realtimeLuceneReadersForRealtimeSegment;
```

Comment:
You can choose to do this in another PR, but does it make sense to call the class MutableLuceneReaders?

Reply:
Done